file/provider/create_snapshot.go

```diff
@@ -0,0 +1,82 @@
+/**
+ * Copyright 2025 IBM Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Package provider ...
+package provider
+
+import (
+	"fmt"
+	"time"
+
+	userError "github.com/IBM/ibmcloud-volume-file-vpc/common/messages"
+	"github.com/IBM/ibmcloud-volume-file-vpc/common/vpcclient/models"
+	"github.com/IBM/ibmcloud-volume-interface/lib/metrics"
+	"github.com/IBM/ibmcloud-volume-interface/lib/provider"
+	"github.com/IBM/ibmcloud-volume-interface/lib/utils/reasoncode"
+	"go.uber.org/zap"
+)
+
+const snapshotReadyState = "stable"
+
+// CreateSnapshot creates snapshot
+func (vpcs *VPCSession) CreateSnapshot(sourceVolumeID string, snapshotParameters provider.SnapshotParameters) (*provider.Snapshot, error) {
+	vpcs.Logger.Info("Entry CreateSnapshot", zap.Reflect("snapshotRequest", snapshotParameters), zap.Reflect("sourceVolumeID", sourceVolumeID))
+	defer vpcs.Logger.Info("Exit CreateSnapshot", zap.Reflect("snapshotRequest", snapshotParameters), zap.Reflect("sourceVolumeID", sourceVolumeID))
+	defer metrics.UpdateDurationFromStart(vpcs.Logger, "CreateSnapshot", time.Now())
+	var err error
+	err = vpcs.validateSnapshotRequest(sourceVolumeID)
+	if err != nil {
+		return nil, err
+	}
+	var snapshotResult *models.Snapshot
+
+	//Convert SnapshotTags map to string[]
+	tags := make([]string, 0, len(snapshotParameters.SnapshotTags))
+	for key, value := range snapshotParameters.SnapshotTags {
+		tags = append(tags, fmt.Sprintf("%s:%s", key, value))
+	}
+
+	snapshotTemplate := &models.Snapshot{
+		Name:     snapshotParameters.Name,
+		UserTags: tags,
+	}
+
+	err = retry(vpcs.Logger, func() error {
+		snapshotResult, err = vpcs.Apiclient.SnapshotService().CreateSnapshot(sourceVolumeID, snapshotTemplate, vpcs.Logger)
+		return err
+	})
+	if err != nil {
```

Comment:
how do you know that its issue with SnapshotSpaceOrderFailed

Reply:
DOne